processor/src/main/java/org/mapstruct/ap/internal/model/source/builtin/JaxbElemToValue.java

```diff
@@ -5,26 +5,23 @@
  */
 package org.mapstruct.ap.internal.model.source.builtin;
 
-import static org.mapstruct.ap.internal.util.Collections.asSet;
-
 import java.util.Set;
 
 import org.mapstruct.ap.internal.model.common.Parameter;
 import org.mapstruct.ap.internal.model.common.Type;
-import org.mapstruct.ap.internal.model.common.TypeFactory;
-import org.mapstruct.ap.internal.util.JaxbConstants;
+
+import static org.mapstruct.ap.internal.util.Collections.asSet;
 
 /**
  * @author Sjaak Derksen
  */
-public class JaxbElemToValue extends BuiltInMethod {
+abstract class JaxbElemToValue extends BuiltInMethod {
 
     private final Parameter parameter;
     private final Type returnType;
     private final Set<Type> importTypes;
 
-    public JaxbElemToValue(TypeFactory typeFactory) {
-        Type type = typeFactory.getType( JaxbConstants.JAXB_ELEMENT_FQN );
+    JaxbElemToValue(Type type) {
```

Comment:
Can't we just change this constructor to get the `JAXBElement` `Type`? This way the class doesn't have to be abstract and the other 2 classes can be removed. We will also have a single `ftl` file.

Reply:
You are right! I can't believe I didn't do it this way in the first place.

Reply:
There was an additional problem as well, the equals method checking the class only.